src/components/data-table-v2/_data-table-v2-action.scss

```diff
@@ -176,3 +176,422 @@
     opacity: 0;
   }
 }
+
+@mixin data-table-v2-action--x {
+  //================================================
+  //TOOLBAR
+  //================================================
+  .#{$prefix}--table-toolbar {
+    display: flex;
+    padding-top: $spacing-xs;
+    padding-bottom: $spacing-xs;
+    width: 100%;
+    position: relative;
+    background: $ui-01;
+    height: rem(48px);
+
+    .#{$prefix}--search-input {
+      position: relative;
+
+      &:focus {
+        box-shadow: inset 0px 0px 0px 1px $brand-01;
+        outline: 0;
+      }
+    }
+  }
+
+  .#{$prefix}--toolbar-content {
+    display: flex;
+    height: rem(48px);
+    position: absolute;
+    top: 0;
+    right: 0;
+  }
+
+  //================================================
+  //HIDDEN SEARCH - DEFAULT TOOLBAR
+  //================================================
+  .#{$prefix}--toolbar-search-container-hidden {
+    height: rem(48px);
+    width: rem(48px);
+    transition: width 175ms $carbon--standard-easing;
+
+    & > .#{$prefix}--search .#{$prefix}--search-input {
+      border-bottom: none;
+      height: rem(48px);
+      @include font-size('14');
+      padding-left: $spacing-3xl;
+      visibility: hidden;
+    }
+
+    & > .#{$prefix}--search .#{$prefix}--search-magnifier {
+      height: rem(48px);
+      width: rem(48px);
+      padding: $spacing-md;
+      left: 0;
+    }
+
+    & > .#{$prefix}--search .#{$prefix}--search-magnifier:focus {
+      @include focus-outline('outline');
+    }
+
+    & > .#{$prefix}--search .#{$prefix}--search-magnifier:hover {
+      background: $hover-field;
+    }
+
+    & > .#{$prefix}--search .#{$prefix}--search-close {
+      height: rem(48px);
+      width: rem(48px);
+    }
+  }
+
+  //================================================
+  //ACTIVE SEARCH - DEFAULT TOOLBAR
+  //================================================
+  .#{$prefix}--toolbar-search-container-active {
+    width: auto;
+    transition: width 175ms $carbon--standard-easing;
+
+    & > .#{$prefix}--search .#{$prefix}--search-input {
+      visibility: visible;
```

Comment:
`visibility: visible` creates an issue inside `visibility: hidden` (ref: https://jakearchibald.com/2014/visible-undoes-hidden/) - `visibility: inherit` is one way to achieve a similar thing.